src/GraphQL/Resolvers/AsyncEventStreamResolver.cs

```diff
@@ -1,44 +1,54 @@
-using GraphQL.Subscription;
-
 namespace GraphQL.Resolvers
 {
-    public class AsyncEventStreamResolver<T> : IEventStreamResolver<T>
+    public class AsyncEventStreamResolver<T> : IEventStreamResolver
     {
-        private readonly Func<IResolveEventStreamContext, Task<IObservable<T?>>> _subscriber;
+        private readonly Func<IResolveFieldContext, ValueTask<IObservable<object?>>> _subscriber;
 
-        public AsyncEventStreamResolver(
-            Func<IResolveEventStreamContext, Task<IObservable<T?>>> subscriber)
+        public AsyncEventStreamResolver(Func<IResolveFieldContext, Task<IObservable<T?>>> subscriber)
         {
-            _subscriber = subscriber ?? throw new ArgumentNullException(nameof(subscriber));
+            if (subscriber == null)
+                throw new ArgumentNullException(nameof(subscriber));
+
+            if (typeof(T).IsValueType)
```

Comment:
Add `where T : class` and remove runtime checks ?

Reply:
See conversation within `EventStreamResolver` below.